test/dlc_tests/conftest.py

```diff
@@ -222,10 +236,201 @@ def pytest_runtest_setup(item):
             pytest.skip("Skipping non-canary tests")
 
 
+def pytest_collection_modifyitems(session, config, items):
+    if config.getoption("--generate-coverage-doc"):
+        failure_conditions = {}
+        test_coverage_file = test_utils.TEST_COVERAGE_FILE
+        test_cov = {}
+        for item in items:
+            # Define additional csv options
+            function_name = item.name.split("[")[0]
+            function_key = f"{item.fspath}::{function_name}"
+            str_fspath = str(item.fspath)
+            str_keywords = str(item.keywords)
+
+            # Construct Category and Github_Link fields based on the filepath
+            category = str_fspath.split('/dlc_tests/')[-1].split('/')[0]
+            github_link = f"https://github.com/aws/deep-learning-containers/blob/master/" \
+                          f"{str_fspath.split('/deep-learning-containers/')[-1]}"
+
+            # Only create a new test coverage item if we have not seen the function before. This is a necessary step,
+            # as parametrization can make it appear as if the same test function is a unique test function
+            if test_cov.get(function_key):
+                continue
+
+            # Based on keywords and filepaths, assign values
+            scope = _infer_field_value("all", ("mxnet", "tensorflow", "pytorch"), str_fspath)
+            train_inf = _infer_field_value("both", ("training", "inference"), str_fspath, str_keywords)
+            integration = _infer_field_value("general integration",
+                                             ("_dgl_", "smdebug", "gluonnlp", "smexperiments", "_mme_", "pipemode",
+                                             "tensorboard", "_s3_"),
+                                             str_keywords)
+            model = _infer_field_value("N/A",
+                                       ("mnist", "densenet", "squeezenet", "half_plus_two", "half_plus_three"),
+                                       str_keywords)
+            num_instances = _infer_field_value(1, ("_multinode_", "_multi-node_"), str_fspath, str_keywords)
+            cpu_gpu = _infer_field_value("all", ("cpu", "gpu", "eia"), str_keywords)
+            if cpu_gpu == "gpu":
+                cpu_gpu = _handle_single_gpu_instances(function_key, str_keywords, failure_conditions)
```

Comment:
update function call to be - `cpu_gpu, failure_conditions = _handle_single_gpu_instances(function_key, str_keywords, failure_conditions)`